src/randomizer.py

```diff
@@ -1412,6 +1412,57 @@ def get_child(parent: Union[ca.StructRef, ca.UnaryOp]) -> ca.Node:
     ensure(changed)
 
 
+def perm_split_assignment(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Split assignments of the form a = b . c . d ...; into a = b; a = a . c . d ...;, a = c . d ...; a = b . a;, etc."""
+    cands = []
+    # Look for assignments of the form 'var = binaryOp' (ignores op=)
+    class Visitor(ca.NodeVisitor):
+        def visit_Assignment(self, node: ca.Assignment) -> None:
+            if node.op == '=' and isinstance(node.rvalue, ca.BinaryOp) and region.contains_node(node):
+                cands.append(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    assign = random.choice(cands)
+    var = assign.lvalue
+
+    binops = []
+
+    def collect_binops(node: ca.BinaryOp) -> None:
+        nonlocal binops
+        if isinstance(node.left, ca.BinaryOp):
+            collect_binops(node.left)
+        binops.append(node)
+        if isinstance(node.right, ca.BinaryOp):
+            collect_binops(node.right)
+
+    collect_binops(typing.cast(ca.BinaryOp, assign.rvalue))
+
+    split = random.choice(binops)
+
+    # Choose which side to move to a new assignment
+    if random.choice([True, False]):
+        side = split.left
+        split.left = copy.deepcopy(var)
+    else:
+        side = split.right
+        split.right = copy.deepcopy(var)
+
+    new_assign = ca.Assignment("=", copy.deepcopy(var), side)
+
+    ins_cands = get_insertion_points(fn, region)
+    ensure(ins_cands)
```

Comment:
Avoid calling `ensure` after modifying the AST, since it will cause some other pass to operate on a damaged AST in a potentially hard to debug manner. `assert` would work I think (i.e. assert False just after the loop below), but it may be safer to move the ins_cands creation and loop through ins_cands to just after `assign` is assigned.